src/scenes/storeScene.js

```diff
@@ -0,0 +1,916 @@
+                if(object_name.indexOf(item) >= 0){
+                  if(this.checklist_done[item].done == false) {
+                    console.log('ajout rature', item)
+                    this.rature = this.add.image(this.checklist_done[item].position_x, this.checklist_done[item].position_y, 'rature')
+                    this.rature.depth = 20;
+                    this.checklist_done[item].done = true;
+                  } else {
+                    console.log('rature deja ajoutée')
+                  }
+                }else{
+                  console.log('pas de', item)
+                }
+              }
+              
+              // Animate basket
+              this.tweens.add({
+                targets: [this.basket, this.basket_front],
+                scale: 1.2,
+                duration: 100,
+                ease: 'Quadratic',
+                yoyo: true,
+                loop: 0
+              })
+          },
+          onCompleteScope: this
+      });
+    }
+
+    create() {
+        this.cameras.main.centerOn(0, 0);
+        this.cameras.main.fadeIn(1000);
+
+        if(this.current_card.isLoaded()) {
+            this.current_card.create();
+            this.addArrow()
+        }
+        
+        this.checklist = this.add.image(-this.cameras.main.width * 0.255, 489, 'liste');
+        this.checklist.depth = 20;
+        
+        
+        this.basket = this.add.image(this.cameras.main.width * 0.24166, 700, 'basket');
+        this.basket.depth = 4;
+        
+        this.basket_front = this.add.image(this.cameras.main.width * 0.24166, 700, 'basket-front');
+        this.basket_front.depth = 25;
+        
+        // Update the saved data
+        // ! Cette scene apparaitra pour deux personnages: independant, etudiant
+        // player.cur_scene = Scenes.STORE;
+    }
+
+    nextCard(){
+        if(this.cardIdx < this.cards.length - 1){
+        
+        // move previous card
+        let container = this.add.container();
+        container.depth = 2;
+        for(let i in this.current_card.children){
+          if(this.current_card.children[i].sprite.depth != 5){
+            container.add(this.current_card.children[i].sprite)
+          }
+        }
+        this.tweens.add({
+          targets: container,
+          x: -1200,
+          duration: 500,
+          ease: 'Quadratic',
+          yoyo: false,
+          loop: 0
+        })
+        
+        this.cardIdx++;
+        this.current_card = this.cards[this.cardIdx];
+        this.current_card.create();
+        if(this.cardIdx >= this.cards.length - 1){
+          console.log('remove arrow')
+          this.nextCardButton.destroy();
+          // ici: prochaine scène
+        }
+      }
+    }
+    
+    nextScene(cardIdx) {
+        // TODO -- depends on current position in story
+        /*this.cameras.main.fadeOut(3000, 0, 0, 0,
+            () => this.scene.start(Scenes.XXXX, { cardIdx: cardIdx }),
+            this
+        );*/
+    }
```

Comment:
You can already send it back to the hallway scene using something like:
```javascript
this.scene.start(Scenes.HALLWAY, {
        cardIdx: HallwayCards.INDEP_GRANDMA,
        damien_gone: false
});
```

Reply:
In the case where damien is shopping, you can send it back to the building scene with something like:
```javascript
 this.scene.start(Scenes.BUILDING, {
        mainMenu: false,
        stage: 3,
        windows: {
            damien: WindowState.OFF,
            grandma: WindowState.OFF,
            family: WindowState.OFF,
            indep: WindowState.OFF
        },
        month: Months.MARCH,
        nextScene: {
            damien: null,
            grandma: null,
            family: null,
            indep: null
        }
    });
```

Reply:
Done